### DISCUSSION TITLE
Composable function with await useAsyncData returns promise

### DISCUSSION THREAD

**Discussion by niklasfjeldberg** [nuxt/nuxt#17086] [Questions]
------------------------------
- Operating System: `Windows_NT`
- Node Version:     `v16.15.0`
- Nuxt Version:     `3.0.0-rc.8`
- Package Manager:  `npm@8.11.0`
- Builder:          `vite`
- User Config:      `ssr`, `modules`, `strapi`, `googleFonts`, `publicRuntimeConfig`, `css`, `vite`
- Runtime Modules:  `@vueuse/nuxt@8.9.4`, `@nuxtjs/strapi@1.5.0`, `@nuxtjs/google-fonts@3.0.0-0`
- Build Modules:    `-`
------------------------------
The issue is that when I try to use an function from an composable that uses `await useAsyncData` it does not return the data.

`/composables/funcs.js`
```
export default function () {
  const globalData = async () => {
    const { find } = useStrapi4();
    const { data } = await useAsyncData('fetch-global-data', () => find('global', { populate: '*' }));
    return data;
  };

  return {
    globalData
  }
}
```

Why does  `globalData` return the functions syntex, and  `globalData()` returns `[Object Promise]` when I use it in an component? How can I fix this?

**Comment by danielroe** [comment]
This might be helpful: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function
